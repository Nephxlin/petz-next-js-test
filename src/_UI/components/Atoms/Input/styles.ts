import styled from "styled-components";


export const InputWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const Input = styled.input`
  padding: 13px 14px;
  border: 1px solid #D5D5D5;
  width:265px;
  border-radius: 8px;
`

export const Label = styled.label`
  font-size: 12px;
  font-weight: 700;
  padding-bottom: 8px;
`

export const Error = styled.span`
  color: red;
`