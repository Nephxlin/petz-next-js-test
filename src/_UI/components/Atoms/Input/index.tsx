import * as S from './styles'

interface ICustomInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  error?: string;
}

const Input = (props: ICustomInputProps) => {
  return (
    <S.InputWrapper data-testid='input-styled-component'>
      {props.label && <S.Label htmlFor={props.id}>{props.label}</S.Label>}
      <S.InputStyled {...props} />
      {props.error && <S.Error data-testid='error-message'>{props.error}</S.Error>}
    </S.InputWrapper>
  )
}

export default Input;