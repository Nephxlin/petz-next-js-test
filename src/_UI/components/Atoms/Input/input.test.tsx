import React from 'react';
import { render, screen } from '@testing-library/react';
import Input from './index';

describe('Input component', () => {
  test('renders Input component with label and no error', () => {
    const { container } = render(<Input label="Username" id="username" />);

    const inputComponent = screen.getByTestId('input-styled-component');
    expect(inputComponent).toBeInTheDocument();

    expect(container.firstChild).toBeInTheDocument();

    const inputElement = screen.getByRole('textbox', { name: 'Username' });
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('id', 'username');

    const errorElement = screen.queryByTestId('error-message');
    expect(errorElement).not.toBeInTheDocument();
    expect(inputComponent).toMatchSnapshot();
  });

  test('renders Input component with error', () => {
    const { container } = render(<Input label="Email" id="email" error="Invalid email address" />);

    const inputComponent = screen.getByTestId('input-styled-component');
    expect(inputComponent).toBeInTheDocument();

    expect(container.firstChild).toBeInTheDocument();

    const inputElement = screen.getByRole('textbox', { name: 'Email' });
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute('id', 'email');


    const errorElement = screen.getByTestId('error-message');
    expect(errorElement).toBeInTheDocument();
    expect(errorElement).toHaveTextContent('Invalid email address');

    expect(inputComponent).toMatchSnapshot();
  });
});
