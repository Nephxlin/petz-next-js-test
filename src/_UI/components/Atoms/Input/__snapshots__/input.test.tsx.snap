// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input component with error 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c2 {
  padding: 13px 14px;
  border: 1px solid #D5D5D5;
  width: 265px;
  border-radius: 8px;
}

.c1 {
  font-size: 12px;
  font-weight: 700;
  padding-bottom: 8px;
}

.c3 {
  color: red;
  font-size: 8px;
}

<div
  class="c0"
  data-testid="input-styled-component"
>
  <label
    class="c1"
    for="email"
  >
    Email
  </label>
  <input
    class="c2"
    error="Invalid email address"
    id="email"
    label="Email"
  />
  <span
    class="c3"
    data-testid="error-message"
  >
    Invalid email address
  </span>
</div>
`;

exports[`Input component renders Input component with label and no error 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
}

.c2 {
  padding: 13px 14px;
  border: 1px solid #D5D5D5;
  width: 265px;
  border-radius: 8px;
}

.c1 {
  font-size: 12px;
  font-weight: 700;
  padding-bottom: 8px;
}

<div
  class="c0"
  data-testid="input-styled-component"
>
  <label
    class="c1"
    for="username"
  >
    Username
  </label>
  <input
    class="c2"
    id="username"
    label="Username"
  />
</div>
`;
